sdk/language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -1080,15 +1095,19 @@ buildIndexTree = foldl' merge empty . map path
       , children = Map.unionWith merge (children t1) (children t2)
       }
 
-processIndexTree :: PackageId -> FilePath -> IndexTree -> IO ()
-processIndexTree pkgId srcDir root = do
+processIndexTree :: PackageId -> PackageReference -> FilePath -> IndexTree -> IO ()
+processIndexTree pkgId pkgRef srcDir root = do
   T.writeFileUtf8 (srcDir </> "index.d.ts") $ T.unlines $
     reexportChildren ES6 root ++
-    [ "export declare const packageId = '" <> unPackageId pkgId <> "';" ]
+    [ "export declare const packageId = '" <> unPackageId pkgId <> "';"
+    , "export declare const packageReference = '" <> forTemplateId pkgRef <> "';"
```

Comment:
Add an extra generated `packageReference` field to the TS/JS modules, which can be used to identify the package if needed.